### PR TITLE
Add demo docs

### LINKED ISSUE #2173: Consider hosting demo docs on the website and possibly migrating them to this repo
Would it make sense to:

- Host the [demo's docs] on the OTel website?
- Migrate the [demo's docs] to this repo?

WDYT @austinlparker @svrnm @cartermp? This would be in the spirit of:

- #833

/cc @open-telemetry/demo-approvers 

[demo's docs]: https://github.com/open-telemetry/opentelemetry-demo/tree/main/docs

### PR DESCRIPTION
**Pull request by chalin**
- Closes #2173 by bringing in the demo docs _as is_, via a git submodule.
- I propose this as a first cut, until the demo docs are frozen, after which we'll be able to migrate them from the demo repo to here. In the meantime, I'll submit some front-matter parameters to the demo repo to change the section weight, add linkTitles, etc.

WDYT @svrnm @cartermp @austinlparker @open-telemetry/demo-approvers ?

**Preview**: https://deploy-preview-2246--opentelemetry.netlify.app/docs/demo

p.s. I know that the scripts are misnamed, but I wanted to keep changes to a minimum in this PR.

### CODE REVIEW

**Review by cartermp — APPROVED** [review]
I like it!

**Comment by svrnm** [comment]
I like it as well! One suggestion: There's a lot of documents below the top level navigation, maybe there should be some folders?

**Comment by chalin** [comment]
> There's a lot of documents below the top level navigation, maybe there should be some folders?

Yes, I can fix that in the demo repo (for the doc sections that already exist) once this lands. WDYT?

**Comment by cartersocha** [comment]
@chalin I think we're comfortable calling the docs stable. @open-telemetry/demo-approvers feel free to chime in otherwise

**Comment by julianocosta89** [comment]
The only thing that doesn't look nice is the architecture page, where we are using Mermaid in GH, but in the docs we have just the text.
Would it be possible to either not add this doc, or render the mermaid in the docs?

**Comment by chalin** [comment]
Thanks for the feedback @cartersocha and @julianocosta89.

> render the mermaid in the docs?

@julianocosta89 - done, see https://deploy-preview-2246--opentelemetry.netlify.app/docs/demo/current_architecture/. Best viewed on a wide display. (I'll try to fix the layout in a followup PR.)

With this change in, can someone from the demo team approve this PR?

**Review by reyang — APPROVED** [review]

**Comment by chalin** [comment]
Thanks @reyang!

I've updated the submodule to refer to the current HEAD. Note that "View page source" etc, won't work until I move the pages into this repo. (I could get them to work, but it isn't worth implementing it for this temporary situation.)

Merging!

**Pull request merged**
